front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatMargin.ts

```diff
@@ -0,0 +1,21 @@
+import type { Margin } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import type { PathStep } from 'reducers/osrdconf/types';
+
+const formatMargin = (pathSteps: PathStep[]): Margin => {
+  const margin: Margin = {
+    boundaries: [],
+    values: [],
+  };
+
+  pathSteps.forEach((step, index) => {
+    if (step.theoreticalMargin !== pathSteps[index - 1]?.theoreticalMargin || index === 0) {
+      if (index !== 0) {
+        margin.boundaries.push(step.id);
+      }
+      margin.values.push(step.theoreticalMargin || 'none');
+    }
```

Comment:
I think it's easier to handle the first step apart, and then all the other steps:
```suggestion
    if (index === 0) {
      margin.values.push(step.theoreticalMargin || 'none');
    } else if (step.theoreticalMargin !== pathSteps[index - 1].theoreticalMargin) {
      margin.boundaries.push(step.id);
      margin.values.push(step.theoreticalMargin || 'none');
    }
```

Reply:
Done, thanks